modules/face/samples/sampleKazemi.cpp

```diff
@@ -0,0 +1,106 @@
+#include<iostream>
+#include<vector>
+#include<fstream>
+#include "opencv2/face/face_alignment.hpp"
+#include "opencv2/core.hpp"
+#include "opencv2/highgui.hpp"
+#include "opencv2/imgproc.hpp"
+#include "opencv2/imgcodecs.hpp"
+using namespace std;
+using namespace cv;
+int main(int argc,char** argv){
+    //Give the file containing the filenames of filenames as argument
+    if(argc!=2){
+        cout<<"No file found";
+        return 0;
+    }
+    string file = argv[1];
+    //cout<<file<<endl;
+    ifstream f(file.c_str());
+    if(!f.is_open()){
+        cout<<"Error while loading file";
+        return 0;
+    }
+    //create a vector to store names of files in which annotations
+    // and image names are found
+    /*The format of the file containing annotations should be of following format
+        /data/abc.jpg
+        123.45,345.65
+        321.67,543.89
+     */
+    vector<string> filenames;
+    string s;
+    while(getline(f,s)){
```

Comment:
For now you are using a file containing filenames. Then that file is processed line by line to process the data. However in OpenCV you can pass a folder with the files. Then call the function `glob()` which reads all filenames in the folder in code. This might make stuff easier to process.

Sample code

```
string location = argv[1];
vector<String> filenames; // this has to be a OpenCV String, not an std::string!
glob(location, filenames);
```